mypy/messages.py

```diff
@@ -819,6 +819,19 @@ def typeddict_instantiated_with_unexpected_items(self,
         self.fail('Expected items {} but found {}.'.format(
             expected_item_names, actual_item_names), context)
 
+    def typeddict_item_name_must_be_string_literal(self,
+                                                   typ: TypedDictType,
+                                                   context: Context):
+        self.fail('Cannot prove expression is a valid item name. Expected one of {}.'.format(
```

Comment:
Style nit: we tend not to use periods in error messages. A more consistent message would be `Cannot prove expression is a valid item name; expected one of [...]` (no dot at the end).

Maybe also truncate the item name list (probably not useful to display dozens of names, for example).